### DISCUSSION TITLE
How to set pressure and stress BCs for THM in porous flow

### DISCUSSION THREAD

**Discussion by Ali1990dashti** [idaholab/moose#25400] [Q&A Modules: General]
Dear all,
I am trying to make a THM simulation for a storage scenario using porous flow. The model is also under the influence of stress in 3 directions, i.e. `x`, `y`, and `z`. For the `x` direction I have set the force on the left side of the model and fixed the `x displacement` to be zero on the right side. For `y`, the force is on the backside while on the front side  `y displacement` is zero. For `z`, the stress BC is on the top while the `z displacement` is zero on the bottom. I have also set three pressure BCs on the left side, back side and right side of the model. When I run the model, the pressure BCs in the surrounding three surfaces of the model drops the pressure value (~ 2MPa) inside the model. How can I avoid it? The following fig shows the simulation in time -29 days (I started from -30 days to maybe reach the equilibrium at time 0):

![THM_PF](https://github.com/idaholab/moose/assets/62764899/837eee19-da65-4026-8f82-9279d70d8488)

The line plot is passing through a depth of 1055 m and I expect a ~ 10 MPa pressure but in the model the pressure is dropping.

My input file and mesh are uploaded to this repository: https://github.com/Ali1990dashti/input_file_for_PF
I very much appreciate any help.

**Comment by GiudGiud** [comment]
@cpgr

**Reply by cpgr** [reply]
I haven't had a chance to have a good look, but something is throwing the initial equilibrium out of whack. Something that stands out is the value of gravity in the `Gravity` kernel - I think your units are Pa not MPa, so it should just be -10?

**Reply by Ali1990dashti** [reply]
Dear @cpgr, thanks for your reply. I changed my mistake regarding the gravity value. I updated it to -9.81 but still the same results. I tried a TH model with almost the same setting in porous flow and it worked, but I have no idea why when I couple the mechanics part to it so many issues happen. On the same face in my mesh I am implementing a force and pressure BC, can this make trouble? I very much appreciate your help.

**Reply by Ali1990dashti** [reply]
@cpgr and @WilkAndy, I am still struggling with the same problem: a THM simulation in porous flow in which the pressure behaves strangely. Thanks to the comment of @cpgr,  I updated the `Gravity` value (to be `-9.81`) but the results did not change.
Thanks for your help and support.
